File: SS/Labs/Lab2/2_comment_identifier.c
// Program to identify whether a given line is a comment or not
#include <stdio.h>
#include <string.h>

int main() {
    char line[100];
    int i = 0;
    
    printf("Enter a line: ");
    fgets(line, sizeof(line), stdin);
    
    // Skip leading spaces
    while (line[i] == ' ' || line[i] == '\t')
        i++;
    
    printf("Result: ");
    
    // Check for single-line comment //
    if (line[i] == '/' && line[i+1] == '/') {
        printf("Single-line comment\n");
    }
    // Check for multi-line comment start /*
    else if (line[i] == '/' && line[i+1] == '*') {
        // Check if it ends on same line
        if (strstr(line, "*/") != NULL)
            printf("Multi-line comment (complete)\n");
        else
            printf("Multi-line comment START\n");
    }
    // Check for multi-line comment end */
    else if (strstr(line, "*/") != NULL) {
        printf("Multi-line comment END\n");
    }
    else {
        printf("NOT a comment\n");
    }
    
    return 0;
}

File: SS/Labs/PythonLabs/Lab2/2_comment_identifier.py
# Lab 2 — Comment Identifier
# Checks whether a given line is a comment (single-line // or multi-line /* */)

def identify_comment(line):
    stripped = line.lstrip()

    if stripped.startswith('//'):
        return "Single-line comment"
    elif stripped.startswith('/*'):
        if '*/' in stripped:
            return "Multi-line comment (complete)"
        else:
            return "Multi-line comment START"
    elif '*/' in line:
        return "Multi-line comment END"
    else:
        return "NOT a comment"

def main():
    line = input("Enter a line: ")
    print("Result:", identify_comment(line))

if __name__ == "__main__":
    main()
